app/components/goodcity/offers-search-overlay.js

```diff
@@ -25,6 +25,53 @@ export default Ember.Component.extend(SearchMixin, {
   },
   searchText: "",
 
+  nameDisplayed(offer) {
+    if (
+      offer.get("firstObject.createdBy") &&
+      offer.get("firstObject.company")
+    ) {
+      this.get("messageBox").custom(
+        this.get("i18n").t("search_offer.offer_select_warning", {
+          offerId: offer.get("firstObject.id"),
+          offerName:
+            offer.get("firstObject.createdBy.firstName") +
+            offer.get("firstObject.createdBy.lastName"),
+          offerCompany: offer.get("firstObject.company.name")
+        }),
+        "Yes",
+        () => {
+          this.send("selectOffer", offer.get("firstObject"));
+        },
+        "No"
+      );
+    } else if (offer.get("firstObject.createdBy")) {
+      this.get("messageBox").custom(
+        this.get("i18n").t("search_offer.offer_select_warning_with_name_id", {
+          offerId: offer.get("firstObject.id"),
+          offerName:
+            offer.get("firstObject.createdBy.firstName") +
+            offer.get("firstObject.createdBy.lastName")
+        }),
+        "Yes",
+        () => {
+          this.send("selectOffer", offer.get("firstObject"));
+        },
+        "No"
+      );
+    } else {
+      this.get("messageBox").custom(
+        this.get("i18n").t("search_offer.offer_select_warning_with_id", {
+          offerId: offer.get("firstObject.id")
+        }),
+        "Yes",
+        () => {
+          this.send("selectOffer", offer.get("firstObject"));
+        },
+        "No"
+      );
+    }
+  },
```

Comment:
When you end up doing a series of `if/else` in a row with big blocks, there's usually simpler/cleaner ways to write it.

For example

```javascript
  nameDisplayed(offer) {
    let data  = { offerId: offer.get('id') };
    let key   = 'search_offer.offer_select_warning_with_id';

    if (offer.get("createdBy")) {
      data['offerName'] = offer.get("createdBy.firstName") + offer.get("createdBy.lastName");
      key = 'search_offer.offer_select_warning_with_name_id';

      if (offer.get("company")) {
        data['offerCompany'] = offer.get("company.name");
        key = 'search_offer.offer_select_warning';
      }
    }

    this.get("messageBox").custom(
      this.get("i18n").t(key, data),
      "Yes",
      () => this.send("selectOffer", offer),
      "No"
    );
  }
```
Also

* `nameDisplay` doesn't tell us much on what it does. Maybe something like `confirmAssignOffer`
* There's no reason to pass an array of offers to a method which only applies to one offer. You can just pass the single offer to the method

Reply:
Thanks for the suggestion @patrixr . Implemented the above said part